public/systems_monitoring_graphs.js

```diff
@@ -0,0 +1,578 @@
+            let brush = d3.brushX()                 // Add the brush feature using the d3.brush function
+                .extent( [ [0,0], [Width, Height] ] ) // initialise the brush area: start at 0,0 and finishes at width,height: it means I select the whole graph area
+                .on("end", updateChart) // Each time the brush selection changes, trigger the 'updateChart' function
+
+            // Create the scatter variable: where both the circles and the brush take place
+            let areaChart = svg.append('g')
+                .attr("clip-path", "url(#clip)")
+
+            // Area generator
+            let area = d3.area().x(d => x(d.datetime)).y0(y(0)).y1(d => y(d.disk_usage.used))
+
+            // Add the brushing
+            areaChart
+                .append("g")
+                .attr("class", "brush")
+                .call(brush);
+
+            // Add the area
+            areaChart.append("path")
+                .datum(data)
+                .attr("class", "memoryArea")  // I add the class memoryArea to be able to modify it later on.
+                .attr("fill", "#0E86D4")
+                .attr("fill-opacity", .6)
+                .attr("stroke", "#0E86D4")
+                .attr("stroke-width", 0.2)
+                .attr("d", area)
+                .on("mouseover", function() {
+                    tooltip.style("display", null);
+                })
+                .on("mouseout", function() {
+                    tooltip.style("display", "none");
+                })
+                .on("mousemove", function(d) {
+                    var xPosition = d3.mouse(this)[0] - 15;//x position of tooltip
+                    var yPosition = d3.mouse(this)[1] - 25;//y position of tooltip
+                    tooltip.attr("transform", "translate(" + xPosition + "," + yPosition + ")");//placing the tooltip
+                    var x0 = x.invert(d3.mouse(this)[0]);//this will give the x for the mouse position on x
+                    var y0 = y.invert(d3.mouse(this)[1]);//this will give the y for the mouse position on y
+                    tooltip.select("text").text(`${d3.timeFormat('%Y-%m-%d (%H:%M)')(x0)} Used: ${decimalFormatter(y0).replace('G', 'GB')}`);//show the text after formatting the date
+                });;
+            
+            let tooltip = svg.append("g")
+                .attr("class", "tooltip")
+                .style("opacity", 1.0)
+                .style("display", "none");
+
+            tooltip.append("rect")
+                .attr("width", 190)
+                .attr("height", 20)
+                .attr("fill", "white")
+                .style("opacity", 1.0);
+
+            tooltip.append("text")
+                .attr("x", 95)
+                .attr("dy", "1.2em")
+                .style("text-anchor", "middle")
+                .attr("font-size", "12px")
+                .attr("font-weight", "bold");
+
+            let idleTimeout
+            function idled() { idleTimeout = null; }
```

Comment:
Could you add a comment explaining what these are for? They're a little bit too cryptic until you read L164.

Reply:
It may also help to say what is being idled here, or call it e.g. transitionDelay.
Idle implies the user is idle i.e. not interacting, but I think all that's going on here is you're waiting before running an animation.

Reply:
Updated thanks. In the code, idleTimeout will hold time set before subsequent brushing trigger `updateChart` function.
i.e.,
```
var returnObj = setTimeout(
    function(){console.log('ping');}, 
    1000
);
console.log(returnObj);

```
When tested with node.js, it returns object with the following property
`{ _idleTimeout: 1000, .... }`
That's why have used idleTimeout cause it contains the value of the second arg to set timeout - The number of milliseconds to wait before executing the code